### DISCUSSION TITLE
$queryRaw vs $queryRawUnsafe

### DISCUSSION THREAD

**Discussion by rtnolan** [prisma/prisma#9271] [Q&A] [answered]
Hi there,

I'm trying to understand why I'm running into an issue with using `$queryRaw`. I have a table that's quite large (~4.6 million rows) I'm trying to execute the following:
```javascript
const results = await this.prisma.$queryRaw<TypeA[]>(
  `SELECT * FROM TypeA WHERE colA='${id}' AND colB > '${date}' AND colC <= '${other}'`
);
```
The PK for the table TypeA looks like (colA, colB, colC, colD) so I'm searching based on 3 of those columns. I keep running into the following error: 
```javascript 
Invalid 'prisma.user.update()' invocation: Server has closed the connection.
```

But when I execute the same query with `$queryRawUnsafe(...)` it works as expected, just wondering what's the difference as I couldn't see from the [docs](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) why one would work in this case and the other wouldn't. Thank you.

**Comment by janpio** [comment, marked as answer]
That is indeed super weird. Can you please create an issue from your discussion, so we can treat this as a bug? `$queryRaw` should not implode here - but it seems the fact that it is a different query causes that to happen for some reason.

**Reply by ryands17** [reply]
@waflerr 👋 

Yes pls create an issue [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=kind%2Fbug&template=bug_report.yml) so that we can look at this.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
